src/time/time.cr

```diff
@@ -567,42 +585,55 @@ struct Time
   # Time.local_offset_in_minutes # => -180
   # ```
   def self.local_offset_in_minutes
-    ifdef linux
-      -LibC.timezone.to_i32 / 60
-    else
-      if LibC.gettimeofday(nil, out tzp) != 0
-        raise Errno.new("gettimeofday")
-      end
-      -tzp.tz_minuteswest.to_i32
+    compute_ticks do |t, offset|
+      offset / Span::TicksPerMinute
     end
   end
 
   protected def self.compute_utc_ticks(ticks)
-    compute_ticks do |t, tp, tz|
-      ticks + tz
+    compute_ticks do |t, offset|
+      ticks - offset
     end
   end
```

Comment:
I don't follow the difference between this one and `utc_ticks` above, and why they're implemented differently.